src/rpc/blockchain.cpp

```diff
@@ -629,8 +635,11 @@ static CBlockUndo GetUndoChecked(BlockManager& blockman, const CBlockIndex& bloc
 
     {
         LOCK(cs_main);
-        if (blockman.IsBlockPruned(blockindex)) {
-            throw JSONRPCError(RPC_MISC_ERROR, "Undo data not available (pruned data)");
+        if (!(blockindex.nStatus & BLOCK_HAVE_DATA)) {
+            if (blockman.IsBlockPruned(blockindex)) {
+                throw JSONRPCError(RPC_MISC_ERROR, "Block not available (pruned data)");
+            }
+            throw JSONRPCError(RPC_MISC_ERROR, "Block not available (not fully downloaded)");
```

Comment:
It feels like the rule of the 3 is being hit here and this code could be DRY'd up? Maybe in a helper function that takes the string `"Block"` or `"Undo"` and does string formatting?

Reply:
I have now added a single function `CheckHeaderForData` that takes a bool for querying undo data.